designs/degraded-nodepools.md

```diff
@@ -0,0 +1,107 @@
+# RFC: Degraded NodePool Status Condition
+
+## Motivation
+
+Karpenter may initiate the creation of nodes based on a NodePool configuration, but these nodes might fail to join the cluster due to unforeseen registration issues that Karpenter cannot anticipate or prevent. An example illustrating this issue is when network connectivity is impeded by incorrect cluster security group configuration, such as missing outbound rule that allows outbound access to any IPv4 address. In such cases, Karpenter will continue its attempts to provision compute resources, but these resources will fail to join the cluster until the outbound rule for the security group is updated. The critical concern here is that users will incur charges for these compute resources despite their inability to be utilized within the cluster. 
```

Comment:
To be clear: I don't think that we're really solving for the problem of cost here -- we're still going to be launching instances and retrying

Reply:
Agreed. Will update.